src/DynamoCoreWpf/Views/PackageManager/PackagePathView.xaml.cs

```diff
@@ -40,54 +42,19 @@ internal PackagePathView(PackagePathViewModel viewModel)
             InitializeComponent();
             this.DataContext = viewModel;
             viewModel.RequestShowFileDialog += OnRequestShowFileDialog;
-            viewModel.PropertyChanged += OnPropertyChanged;
-            UpdateVisualToReflectSelectionState();
-            PreviewKeyDown += OnPackagePathDialogKeyDown;
         }
 
         #endregion
 
         #region Private Helper Methods and Event Handlers
-
-        void OnPathSelectionChanged(object sender, SelectionChangedEventArgs e)
-        {
-            if (e.AddedItems.Count <= 0)
-                return; // Nothing selected.
-
-            var selected = e.AddedItems[0] as string;
-            ViewModel.SelectedIndex = ViewModel.RootLocations.IndexOf(selected);
-        }
-
-        private void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
-        {
-            if (e.PropertyName.Equals("SelectedIndex"))
-            {
-                // Repositioning the selection should retain its visual state.
-                UpdateVisualToReflectSelectionState();
-            }
-        }
-
-        private void OnEllipsisClicked(object sender, MouseButtonEventArgs e)
+        private void PackagePathView_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
```

Comment:
Is this for the add paths functionality?

Reply:
yes exactly, when the data context is set to the PackagePathViewModel, we need to hook up the handler to open the dialog.